### ISSUE TITLE
Scroll snap point

### ISSUE THREAD

**Issue by Akxe** [angular/components#16038]
#### What is the expected behavior?
Setting CSS `scroll-snap-type: both mandatory;` should enable scroll snapping.

#### What is the current behavior?
No scroll snapping ever happens.

#### What are the steps to reproduce?
https://stackblitz.com/angular/dnnjdjbpaaag?file=app%2Fcdk-virtual-scroll-horizontal-example.css

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
"@angular/cdk": "7.3.7",
"@angular/core": "7.2.15",
"typescript": "~3.1.1"
Browsers: all

#### Is there anything else we should know?
Closely related to #14833 but it could be fixed other way too.

**Comment by Akxe** [comment]
@andrewseguin Can I help on this? I suppose the solution would be to either spawn N 1x1px divs scaled accordingly to its data, possibly collapsing those that are out of the screen.

**Comment by aneeskodappana** [comment]
any updates on this issue @Akxe Akxe

**Comment by Akxe** [comment]
@aneeskodappana not really, there are some apis, that might offer better ways to do virtual scrolling, but as of now. There is no way around it

**Comment by fredspivock** [comment]
Is there any workaround for this? Would be awesome! Thank you for the project, its a real lifesaver.

**Comment by Akxe** [comment]
The best you can do now is to use `*ngFor` and the `content-visibility: auto; contain-intrinsic-size: height width;` combo.

**Comment by mark922** [comment]
Any updates on the issue? This would be really interesting feature.

**Comment by pranavkumar389** [comment]
Any workaround for supporting this?
